app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -377,23 +380,29 @@ class ProjectInformationActivity : AppCompatActivity() {
         }
     }
 
-    private fun handleVideoWithFavoritePersistenceFiltering(videosLinks: List<String>){
-        //get the favorite list and when it's available provide it to the function that is
-        //responsible for downloading videos in volatile or non volatile memory
+    private fun handleVideo(videosLinks: List<String>){
+        //TODO : change the firebase Auth get instance to hilt dependency injection when possible
+        //get the favorite list, the applied To list and the own projects list
         userdataDatabase.getListOfFavoriteProjects({ favorites ->
-            addVideoAfterDownloadedWithFavoritePersistence(videosLinks, favorites.contains(projectVar.id))
+            projectDatabase.getAllProjects({projects ->
+                addVideoDownload(videosLinks, favorites.contains(projectVar.id),
+            projects.filter { project -> project.authorId == FirebaseAuth.getInstance().uid })},
+                {})
+
         },{
 
         })
     }
 
     // download all videos and add them to the video player
-    private fun addVideoAfterDownloadedWithFavoritePersistence(videosLinks: List<String>, isFavorite : Boolean ) {
+    private fun addVideoDownload(videosLinks: List<String>, isFavorite : Boolean, ownProjects: List<ImmutableProject> ) {
 
         videosLinks.forEach { link ->
-            if(isFavorite){//storing the video with persistence
+            if(isFavorite || ownProjects.any {project -> project.id == projectVar.id}){//storing the video in the permanent memory
+                deletingAndCopyingVideo(link, cacheDir, projectDir)
                 storingVideo(link, projectDir)
             }else {
+                deletingAndCopyingVideo(link, projectDir, cacheDir)//storing
                 storingVideo(link, cacheDir)
             }
```

Comment:
This code will try to copy/delete file each time a video is loaded, even if it was already in the correct directory.
Is it the expected behavior ?
And the if-else block seems nearly identical, maybe it could be use as a single block with 2 val for from/to directory ?

Reply:
the code doesn't move the two files if the file that needs to be moved doesn't exist if(file.exist()) in function

It can be put with intermediate val from/to but I still need the if else block, I don't feel like creating two extra intermediate vals makes the code better